### DISCUSSION TITLE
opengraph-image file url is NOT right, always start with localhost:3000

### DISCUSSION THREAD

**Discussion by jkiss** [vercel/next.js#53566] [Help] [answered]
### Summary

opengraph-image file's url always start with localhost:3000 after nextjs build, I can not find how to set it's host, need help!

**Comment by Nadeera3784** [comment]
Hi @jkiss ,

Did you set the basePath in next.config.js?
```
module.exports = {
  // Replace 'your-app-url' with your actual production URL.
  basePath: process.env.NODE_ENV === 'production' ? '/your-app-url' : '',
};
```
In your code where you set the OpenGraph image URL, use the base URL you defined in next.config.js as the prefix for the image path. For example:

```
// Replace '/path/to/your/image.jpg' with the actual path to your image.
const ogImageUrl = `${process.env.NODE_ENV === 'production' ? '/your-app-url' : ''}/path/to/your/image.jpg`;

```

**Comment by jkiss** [comment, marked as answer]
there is a `metadataBase` option, but it only works build&start that very confusing me 🤣

**Comment by Josehower** [comment]
As reference this is the link to the solution https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadatabase
